broadinstitute/single_cell_portal_core
# Fixing boot command to correctly pass Docker version tag (SCP-4578)

#### BACKGROUND
The first deployment using our new [Docker image infrastructure](https://github.com/broadinstitute/single_cell_portal_core/pull/1552) encountered an issue where the release version tag was not properly passed to the remote boot command, resulting in the `development` image being used, rather than the release tag of `1.23.0`.  This was not a functional issue, as the images were (at the time) identical, but moving forward this can potentially cause hotfixes/rollbacks to fail as they would not properly honor the image tag.

#### CHANGES
The issue was apparently in splitting a bash command over two lines, but then passing that command to a separate function that would run the command in the project root directory.  Because the command was tokenized and passed with `"$1"`, only the first half was transmitted, which trims off the `$VERSION_TAG`.  The rest of the dropped parameters were present inside the .env secrets file, and were still passed to the Docker container.  This update removes the line split on the command, and also adds some additional logging to allow the release engineer to see which tag has been used at each stage.

#### MANUAL TESTING
1. Pull this branch and ensure Docker is running locally
2. Run `./rails_local_setup.rb` to write out your config file locally
3. Run the `remote_deploy.sh` script from the project root with the following parameters:
```
DESTINATION_BASE_DIR=$(pwd)
PORTAL_SECRETS_PATH=./config/secrets/.source_env.bash
DESTINATION_BASE_DIR=$DESTINATION_BASE_DIR PORTAL_SECRETS_PATH=$PORTAL_SECRETS_PATH PASSENGER_APP_ENV=development bin/remote_deploy.sh -v 1.23.0
```
4. Ensure that the portal boots with the correct tag (you don't need to let it complete, just look for the following log messages:
```
1.23.0: Pulling from broad-singlecellportal-staging/single-cell-portal
...
gcr.io/broad-singlecellportal-staging/single-cell-portal:1.23.0
### COMPLETED ###
### loading env secrets ###
### COMPLETED ###
### Booting single_cell with Docker image version 1.23.0 ###
### BOOTING PORTAL IN 'development' MODE WITH DOCKER IMAGE VERSION 1.23.0
*** Running /etc/my_init.d/00_regen_ssh_host_keys.sh...
*** Running /etc/my_init.d/01_set_user_permissions.bash...
*** Running /etc/my_init.d/02_generate_dh_parameters.bash...
generating diffie-hellman parameters...
Generating DSA parameters, 4096 bit long prime
```
